clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslServer.java

```diff
@@ -84,13 +87,16 @@ public OAuthBearerSaslServer(CallbackHandler callbackHandler) {
             throw new SaslAuthenticationException(errorMessage);
         }
         errorMessage = null;
+
         OAuthBearerClientInitialResponse clientResponse;
         try {
             clientResponse = new OAuthBearerClientInitialResponse(response);
         } catch (SaslException e) {
             log.debug(e.getMessage());
             throw e;
         }
+        extensions = clientResponse.extensions();
```

Comment:
It is possible that the extensions could be set and then the process() method either returns an error response to the client or throws an exception.  It probably isn't too much of an issue if this happens, but best to set the extensions at the same point where tokenForNegotiatedProperty is set, so probably should add the extensions as a new parameter on the process() call.

Reply:
Good catch